### ISSUE TITLE
The definition for "kreait\firebase\factory" has no class.

### ISSUE THREAD

**Issue by iamakanksha** [kreait/firebase-bundle#26]
## The problem
Installed the bundle (version 1.2) using composer. Added the bundle to the AppKernel.php and pushed the commit to GitHub. My build is failing due to the following error:
```
[Symfony\Component\DependencyInjection\Exception\RuntimeException]                                                                                                         
  The definition for "kreait\firebase\factory" has no class. If you intend to inject this service dynamically at runtime, please mark it as synthetic=true. If this is an a  
  bstract definition solely used by child definitions, please add abstract=true, otherwise specify a class to get rid of this error. 

```
## Environment
- PHP Version: 
7.1.33 
- Symfony version:
2.8.52
- Firebase PHP SDK and Bundle version:
kreait/firebase-bundle                 1.2.0                        Symfony Bundle for the Firebase Admin SDK
kreait/firebase-php                    4.18.0                       Firebase Admin SDK
kreait/firebase-tokens                 1.7.2                        A library to work with Firebase tokens
kreait/gcp-metadata                    1.0.1                        Get the metadata from a Google Cloud Platform environment.

## Details
Tried to set up the bundle with the project, by registering it in AppKernel.php, which is one of the initial steps and it is failing at that point.

## Code to reproduce issue 
Inside app/AppKernel.php
```

// ...
class AppKernel extends Kernel
{
    // ...

    public function registerBundles()
    {
        $bundles = array(
            // ...
           new Kreait\Firebase\Symfony\Bundle\FirebaseBundle()
        );
        // ...
    }
}
```

**Comment by jeromegamez** [comment]
I believe you need to write `new \Kreait` instead of `new Kreait` (prepend a backslash)

**Comment by iamakanksha** [comment]
> I believe you need to write `new \Kreait` instead of `new Kreait` (prepend a backslash)

I tried this out but the issue remains

**Comment by jeromegamez** [comment]
1.2 is a rather old release of the bundle, it might be a bug. Is there a reason why newer 1.x versions of the bundle don't work?

**Comment by iamakanksha** [comment]
> 1.2 is a rather old release of the bundle, it might be a bug. Is there a reason why newer 1.x versions of the bundle don't work?

My project uses symfony version 2.8, and firebase-bundle 1.2 seems like the last version compatible with the symfony version used by the project

**Comment by jeromegamez** [comment]
Have you added code to your application using the SDK, or does the error occur already only by adding the bundle?

Unfortunately I don't have the capacities at the moment to look into it (and Symfony 2.x has been out of support for quite some time now), that's why I currently only can ask question to help guessing what the issue might be.

Given that it's broken for you, it's totally possible that the bundle is generally broken for Symfony 2.x. Perhaps the best solution would be to take the bundle as a kind of blueprint and implement it yourself specifically for your project 🤔

**Comment by iamakanksha** [comment]
> Have you added code to your application using the SDK, or does the error occur already only by adding the bundle?
> 
> Unfortunately I don't have the capacities at the moment to look into it (and Symfony 2.x has been out of support for quite some time now), that's why I currently only can ask question to help guessing what the issue might be.
> 
> Given that it's broken for you, it's totally possible that the bundle is generally broken for Symfony 2.x. Perhaps the best solution would be to take the bundle as a kind of blueprint and implement it yourself specifically for your project 🤔

The error occurs only on adding the bundle to the project

**Comment by iamakanksha** [comment]
Tried to update my firebase-php version to a higher version that supports Symfony 2.8 so that I could update the firebase-bundle version. I directly tried to update the firebase-php to version 4.35 to be able to use version the bundle version 1.7, however the installation itself failed. I tried to update the firebase-php version to 4.22 so that I could upgrade to bundle version 1.3, but this issue remains in that version.

**Comment by iamakanksha** [comment]
@jeromegamez, figured a fix for it however I am unable to create a PR

**Comment by iamakanksha** [comment]
@jeromegamez I am unable to create a PR on the older tag of the repository (1.2.0)

**Comment by anujith-singh** [comment]
@jeromegamez the fix for this issue is https://github.com/kreait/firebase-bundle/compare/1.2.0...anujith-singh:fix_symfony_2_8?expand=1

But how do we create a pull request and merge these changes!

**Comment by jeromegamez** [comment]
I just released [1.2.1](https://github.com/kreait/firebase-bundle/releases/tag/1.2.1) with [your changes](https://github.com/kreait/firebase-bundle/compare/1.2.0...1.2.1). Happy coding!
